### PR TITLE
feat(node/crypto): add `scrypt` and `scryptSync`

### PR DESCRIPTION
**Pull request by talentlessguy**
Implementation borrowed from https://github.com/crypto-browserify/scrypt

- [x] I have added tests
- [x] I have run `deno fmt` and `deno lint`

### CODE REVIEW

**Comment by bartlomieju** [comment]
> Implementation borrowed from https://github.com/crypto-browserify/scrypt

There should be relevant copyrights included in the source.

**Comment by talentlessguy** [comment]
@bartlomieju ok one sec will add a license comment

**Comment by talentlessguy** [comment]
done, @bartlomieju

File: node/_crypto/scrypt_test.ts [hunk 1]
```diff
@@ -0,0 +1,179 @@
+import { scrypt, scryptSync } from "./scrypt.ts";
```
Comment (kt3k) [inline node/_crypto/scrypt_test.ts:1]: nit: This file needs copyright header

```
// Copyright 2018-2021 the Deno authors. All rights reserved. MIT license.
```

**Review by kt3k — APPROVED** [review]
LGTM with nits. Nice work! @talentlessguy 

Also confirmed the test cases match with Node.js results.

File: node/_crypto/scrypt_test.ts (hunk 1 above)
Comment (kt3k) [reply node/_crypto/scrypt_test.ts:1]: Let's address this in another PR as we already have many files missing copyright headers.

Comment (talentlessguy) [reply node/_crypto/scrypt_test.ts:1]: ok no problem

and thanks for merging!

deno_std/node/crypto is getting closer to node.js :D

**Pull request merged**